src/cryptonote_protocol/cryptonote_protocol_handler.inl

```diff
@@ -807,9 +809,10 @@ namespace cryptonote
         return 1;
       }
       //to avoid concurrency in core between connections, suspend connections which delivered block later then first one
+      const crypto::hash block_hash = get_block_hash(b);
       if(count == 2)
       {
-        if(m_core.have_block(get_block_hash(b)))
+        if(m_core.have_block(block_hash))
```

Comment:
Should the check above be `count == arg.blocks.size()` ? The current behavior is a bit odd, if the second block is in the core it skips everything below ...